test/JDBC/expected/exec_sp_in_udf-vu-verify.out

```diff
@@ -0,0 +1,206 @@
+begin
+exec sp_executesql 'select 123'
+return 0
+end
+go
+select dbo.f1_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Only functions can be executed within a function)~~
+
+
+create function f2_exec_sp_in_udf() 
+returns int as
+begin
+exec dbo.sp_helpdb
+return 0
+end
+go
+select dbo.f2_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Only functions can be executed within a function)~~
+
+
+create function f3_exec_sp_in_udf() 
+returns int as
+begin
+exec [sp_helpdb]
+return 0
+end
+go
+select dbo.f3_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Only functions can be executed within a function)~~
+
+
+create function f4_exec_sp_in_udf() 
+returns @tv table(a int) as
+begin
+exec sp_myproc
+return
+end
+go
+select * from dbo.f4_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Only functions can be executed within a function)~~
+
+
+create function f5_exec_sp_in_udf() 
+returns @tv table(a int) as
+begin
+exec dbo.[sp_myproc]
+return
+end
+go
+select * from dbo.f5_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Only functions can be executed within a function)~~
+
+
+create function f6_exec_sp_in_udf() 
+returns @tv table(a int) as
+begin
+exec tempdb..p_tempdb_exec_sp_in_udf
+return
+end
+go
+select * from dbo.f6_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Only functions can be executed within a function)~~
+
+
+
+-- multi-statement TVF cannot be called with EXEC
+exec dbo.f_tvf_exec_sp_in_udf
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: master_dbo.f_tvf_exec_sp_in_udf() is not a procedure)~~
+
+create function f7_exec_sp_in_udf() 
+returns int
+begin
+exec dbo.f_tvf_exec_sp_in_udf
+return
+end
+go
+select * from dbo.f7_exec_sp_in_udf()
+go
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: master_dbo.f_tvf_exec_sp_in_udf() is not a procedure)~~
+
+
+-- inline TVF should not be callable with EXEC, but Babelfish currently allows this
+-- when this gets fixed so that an error is raised, the result of this test will change
+exec dbo.f_itvf_exec_sp_in_udf
+go
+create function f8_exec_sp_in_udf() 
+returns int
+begin
+exec dbo.f_itvf_exec_sp_in_udf
+return
+end
+go
+select * from dbo.f8_exec_sp_in_udf()
+go
+~~START~~
+int
+<NULL>
+~~END~~
+
+
+
+-- procedure and trigger are not affected:
+create trigger tr1 on t1_exec_sp_in_udf for insert as
+begin
+	exec sp_executesql 'select 123 as in_trigger'
+end
+go
+insert t1_exec_sp_in_udf values(456)
+go
+~~START~~
+int
+123
+~~END~~
+
+~~ROW COUNT: 1~~
+
+select * from t1_exec_sp_in_udf
+go
+~~START~~
+int
+456
+~~END~~
+
+
+create procedure p1_exec_sp_in_udf
+as
+begin
+	exec sp_executesql 'select 123 as in_procedure'
+end
+go
+exec p1_exec_sp_in_udf
+go
+~~START~~
+int
+123
+~~END~~
+
+
+-- scalar UDF can still be called with EXEC
+declare @v int
```

Comment:
Can we add the following test case as well:
```
1> declare @a int
2> set @a = dbo.f1_exec_sp_in_udf()
3> go
Msg 557, Level 16, State 2, Server EC2AMAZ-EULFEOJ, Line 2
Only functions and some extended stored procedures can be executed from within a function.
```

Reply:
added this case plus various other variations